From e66bb8f6de328cc95d3b0ae39a5ccc37b6b19617 Mon Sep 17 00:00:00 2001
From: Biswapriyo Nath <nathbappai@gmail.com>
Date: Sat, 9 Aug 2025 15:03:24 +0000
Subject: [PATCH] plaintextextractor: Fix clang compiler error for 32 bit
 architectures

This commit fixes the following compiler errors.

plaintextextractor.cpp:124:57: error: non-constant-expression cannot be narrowed
from type 'qint64' (aka 'long long') to 'qsizetype' (aka 'int') in initializer list [-Wc++11-narrowing]
plaintextextractor.cpp:126:48: error: non-constant-expression cannot be narrowed
from type 'qint64' (aka 'long long') to 'qsizetype' (aka 'int') in initializer list [-Wc++11-narrowing]
plaintextextractor.cpp:163:45: error: non-constant-expression cannot be narrowed
from type 'qint64' (aka 'long long') to 'qsizetype' (aka 'int') in initializer list [-Wc++11-narrowing]
---
afaics, the narrowing is not a problem because the size of the buffer used (256
* 1024 = 262_144 bytes) is well below the 32-bit maximum of ~4 billion, so the
narrowing will never be a problem

 src/extractors/plaintextextractor.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/extractors/plaintextextractor.cpp b/src/extractors/plaintextextractor.cpp
index bcd07e28..9458855d 100644
--- a/src/extractors/plaintextextractor.cpp
+++ b/src/extractors/plaintextextractor.cpp
@@ -119,7 +119,7 @@ void PlainTextExtractor::extract(ExtractionResult* result)
 
     // Read the first chunk, detect the encoding and decode it
     QByteArray chunk(256 * 1024, Qt::Uninitialized);
-    auto size = file.read(chunk.data(), chunk.size());
+    qsizetype size = file.read(chunk.data(), chunk.size());
 
     QStringDecoder codec{autodetectCodec({chunk.data(), size})};
 
@@ -151,7 +151,7 @@ void PlainTextExtractor::extract(ExtractionResult* result)
 
     // Read and decode the remainder
     while (!file.atEnd()) {
-        auto size = file.read(chunk.data(), chunk.size());
+        qsizetype size = file.read(chunk.data(), chunk.size());
         if (size < 0) {
             // may happen when the file is truncated during read
             qCWarning(KFILEMETADATA_LOG) << "Error reading" << result->inputUrl();
-- 
GitLab

